hazelcast/src/main/java/com/hazelcast/jet/impl/execution/TaskletExecutionService.java

```diff
@@ -392,11 +393,15 @@ private void runTasklet(TaskletTracker t) {
                 progressTracker.mergeWith(result);
             } catch (Throwable e) {
                 if (e instanceof CancellationException) {
-                    logger.info("Job was cancelled by the user.");
+                    logger.fine("Job was cancelled by the user.");
                     CancellationException ex = (CancellationException) e;
                     t.executionTracker.exception(ex);
+                } else if (e instanceof ResultLimitReachedException) {
+                    logger.fine("SQL LIMIT is reached. ");
+                    ResultLimitReachedException ex = (ResultLimitReachedException) e;
+                    t.executionTracker.exception(ex);
                 } else {
-                    logger.warning("Exception in " + t.tasklet, e);
+                    logger.info("Exception in " + t.tasklet, e);
```

Comment:
We should do the same error handling for BlockingWorker. We can extract this code to a method and call it from there. Currently, the blocking worker logs in WARN level.

Reply:
Do you mean we should check for the same exceptions there as well? Or just the logging level? @viliam-durina

Reply:
Exceptions, same log level. In principle it should be the same, whether we're running as cooperative or blocking.